contrib/kind.sh

```diff
@@ -531,12 +546,38 @@ set_default_params() {
   JOIN_SUBNET_IPV4=${JOIN_SUBNET_IPV4:-100.64.0.0/16}
   JOIN_SUBNET_IPV6=${JOIN_SUBNET_IPV6:-fd98::/64}
   KIND_NUM_MASTER=1
+  OVN_ENABLE_INTERCONNECT=${OVN_ENABLE_INTERCONNECT:-false}
+
+  if [ "$OVN_HA" == true ] && [ "$OVN_ENABLE_INTERCONNECT" != false ]; then
+     echo "HA mode cannot be used together with Interconnect"
+     exit 1
+  fi
+
+  if [ "$OVN_COMPACT_MODE" == true ] && [ "$OVN_ENABLE_INTERCONNECT" != false ]; then
+     echo "Compact mode cannot be used together with Interconnect"
+     exit 1
+  fi
+
   if [ "$OVN_HA" == true ]; then
     KIND_NUM_MASTER=3
     KIND_NUM_WORKER=${KIND_NUM_WORKER:-0}
   else
     KIND_NUM_WORKER=${KIND_NUM_WORKER:-2}
   fi
+
+  if [ "$OVN_ENABLE_INTERCONNECT" == false ]; then
+    KIND_NUM_ZONES=1
+  else
+    # Default is 3 zones (ovn-control-plane, ovn-worker, ovn-worker2)
+    KIND_NUM_ZONES=${KIND_NUM_ZONES:-3}
+    # Lets support only node per zone.
+    KIND_NUM_NODES_PER_ZONE=1
+
+    KIND_TOTAL_NODES=$((KIND_NUM_ZONES * KIND_NUM_NODES_PER_ZONE))
+    KIND_NUM_MASTER=1
+    KIND_NUM_WORKER=$((KIND_TOTAL_NODES - 1))
```

Comment:
--num-workers is confusing when multi zone is enabled.

We have a new option called -- KIND_NUM_NODES_PER_ZONE.

If KIND_NUM_NODES_PER_ZONE is 1, then each node is a zone and we deploy ovnkube-node pod with the entire OVN stack and ovnkube-controller and ovnkube-node.

If KIND_NUM_NODES_PER_ZONE is > 1, then it is a multiple zone with multiple nodes per zone deployment.

If some one passes --num-workers we just ignore it.

Sure from --num-workers we can derive number of nodes per zone as  num-zones / num-workers.

But this could result in odd nodes.

I guess its fine to just ignore num-workers for multi zone deployments.

Reply:
OK, I would guess that if someone wanted an even number of nodes across zones they could just work out the math.

Maybe we can add to the help output that num-workers is ignored if enable-interconnect is used?